numpy/core/src/multiarray/dtypemeta.c

```diff
@@ -122,6 +124,50 @@ legacy_dtype_default_new(PyArray_DTypeMeta *self,
     return (PyObject *)self->singleton;
 }
 
+static PyObject *
+string_unicode_new(PyArray_DTypeMeta *self, PyObject *args, PyObject *kwargs)
+{
+    npy_intp size;
+
+    static char *kwlist[] = {"", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O&", kwlist,
+                                     PyArray_IntpFromPyIntConverter, &size)) {
+        return NULL;
+    }
+
+    if (size < 0) {
+        PyErr_Format(PyExc_ValueError,
```

Comment:
Changed this to value error, because I thought even the other ones make more sense that way.  But then, it shares the error with other paths, so only did this one.

If anyone doesn't like, happy to follow up.  Will merge once tests are done, thanks Nathan.